driver-core/src/main/com/mongodb/client/model/densify/DensifyOptions.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.client.model.densify;
+
+import com.mongodb.annotations.Evolving;
+import com.mongodb.client.model.Aggregates;
+import com.mongodb.lang.Nullable;
+import org.bson.conversions.Bson;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+
+/**
+ * Represents optional fields of the {@code $densify} pipeline stage of an aggregation pipeline.
+ *
+ * @see Aggregates#densify(String, DensifyRange, DensifyOptions)
+ * @see Aggregates#densify(String, DensifyRange)
+ * @mongodb.server.release 5.1
+ * @since 4.7
+ */
+@Evolving
+public interface DensifyOptions extends Bson {
+    /**
+     * Creates a new {@link DensifyOptions} with the specified {@code fields} to partition by.
+     *
+     * @param fields The fields to partition by.
+     * @return A new {@link DensifyOptions}.
+     * @mongodb.driver.manual core/document/#dot-notation Dot notation
```

Comment:
Is this the right link its not obvious how its related

Reply:
This is a documentation for `partitionByFields`, which accepts fields, and this link suggests that the dot-notation is allowed (in some places we allow the dot-notation for fields, and in some we don't).

Reply:
Perhaps mention dot notation is supported in the docs above?